components/TabButton/index.js

```diff
@@ -0,0 +1,17 @@
+import { StyledButton } from "./TabButton.styled";
+
+export default function TabButton({ children, isActive, onClick }) {
+  if (isActive) {
+    return <StyledButton $isActive={isActive}>{children}</StyledButton>;
+  }
+  return (
+    <StyledButton
+      $isActive={isActive}
+      onClick={() => {
+        onClick();
+      }}
+    >
+      {children}
+    </StyledButton>
+  );
```

Comment:
If I'm not mistaken about your intent, this is the same but shorter.
```suggestion
  return (
    <StyledButton
      $isActive={isActive}
      onClick={onClick ? onClick : null}
    >
      {children}
    </StyledButton>
  );
```